### DISCUSSION TITLE
USB mouse code for RP2040?

### DISCUSSION THREAD

**Discussion by FlydGitHub** [micropython/micropython#11608] [RP2040 / Pico]
Does anyone had code for a USB mouse  RP2040?

Any suggestions appraised.

**Comment by jimmo** [comment]
Do you want the Pico to be a USB mouse (i.e. device mode), or to connect a USB mouse to the Pico (i.e. host mode).

Neither are currently supported, but the former is work-in-progress, and _is_ already supported on CircuitPython.

The latter is something we'd like to support eventually.

**Reply by jimmo** [reply]
I should add that usb device mode, and HID in particular (i.e. keyboard/mouse) is supported on the stm32 port (including pyboard). Just not (yet) on others (i.e. Pico, SAMD, etc).

**Comment by FlydGitHub** [comment]
If I unbiased correctly I do indeed want the Pico to be a USB mouse (i.e. device mode)
Thanks Jimmo for your understudying.....
